jina/peapods/pod.py

```diff
@@ -82,15 +80,27 @@ def _parse_args(self, args: Namespace) -> Dict[str, Optional[Union[List[Namespac
             'tail': None,
             'peas': []
         }
-
         if getattr(args, 'parallel', 1) > 1:
             # reasons to separate head and tail from peas is that they
             # can be deducted based on the previous and next pods
             peas_args['head'] = _copy_to_head_args(args, args.polling.is_push)
             peas_args['tail'] = _copy_to_tail_args(args)
             peas_args['peas'] = _set_peas_args(args, peas_args['head'], peas_args['tail'])
+            self.is_head_router = True
+            self.is_tail_router = True
+        elif getattr(args, 'uses_before', None) or getattr(args, 'uses_after', None):
+            args.scheduling = SchedulerType.ROUND_ROBIN
+            if getattr(args, 'uses_before', None):
+                peas_args['head'] = _copy_to_head_args(args, args.polling.is_push)
+                self.is_head_router = True
```

Comment:
What about `self.is_tail_router` in case, `uses_after` is not set, but only `uses_before`? Should it be set explicitly to `False`?

Reply:
please note `__init__` already has `is_head_router ` `is_tail_router ` set